src/main/java/org/matsim/mosaik2/agentEmissions/PositionEmissionsToCSVHandler.java

```diff
@@ -0,0 +1,112 @@
+package org.matsim.mosaik2.agentEmissions;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.log4j.Logger;
+import org.matsim.api.core.v01.events.Event;
+import org.matsim.contrib.emissions.PositionEmissionsModule;
+import org.matsim.core.api.experimental.events.EventsManager;
+import org.matsim.core.events.EventsUtils;
+import org.matsim.core.events.MatsimEventsReader;
+import org.matsim.core.events.handler.BasicEventHandler;
+import org.matsim.core.utils.io.IOUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class PositionEmissionsToCSVHandler implements BasicEventHandler {
+    private static final Logger log = Logger.getLogger(PositionEmissionsToCSVHandler.class);
+
+    final String separator = ";";
+    private final String[] HEADER = {"time", "vehicle_id", "x", "y", "CO",
+            "CO2", "NOx", "NO2", "PM10", "NO"};
+    private final List<List<String>> row2columns = new ArrayList<>();
+
+
+    public static void main(String[] args) {
+        PositionEmissionsToCSVHandler.Input input = new PositionEmissionsToCSVHandler.Input();
+        JCommander.newBuilder().addObject(input).build().parse(args);
+        log.info("Input events file: " + input.eventsFile);
+        log.info("Output csv file: " + input.outputCsvFile);
+
+        PositionEmissionsToCSVHandler csvHandler = new PositionEmissionsToCSVHandler();
+        EventsManager events = EventsUtils.createEventsManager();
+        events.addHandler(csvHandler);
+
+        MatsimEventsReader reader = new MatsimEventsReader(events);
+        events.initProcessing();
+        reader.readFile(input.eventsFile);
+        events.finishProcessing();
+
+        csvHandler.print(input.outputCsvFile);
```

Comment:
This is all good and looks like it would work. Just a general remark: 

Since you don't know the size of an events file. You could also open the csv printer when you create the handler and append the events to the csv file while they are coming in. This way you wouln't need to store all the data in your row2columns list.